modules/renderer/features.js

```diff
@@ -529,17 +537,39 @@ export function rendererFeatures(context) {
 
 
     features.isHidden = function(entity, resolver, geometry) {
-        if (!_hidden.length) return false;
         if (!entity.version) return false;
-
+        if (!features.featureFitsDateRange(entity)) return true;
+        if (!_hidden.length) return false;
         var fn = (geometry === 'vertex' ? features.isHiddenChild : features.isHiddenFeature);
         return fn(entity, resolver, geometry);
     };
 
 
-    features.filter = function(d, resolver) {
-        if (!_hidden.length) return d;
+    features.featureFitsDateRange = function (entity) {
+        if (!context.features().dateRange) return true; // no Date Range e.g. unit tests
+
+        // entity's start & end date
+        // filtering date range from the on-screen controls
+        const entityRange = {
+            'start_date': utilNormalizeDateString(entity.tags.start_date),
+            'end_date': utilNormalizeDateString(entity.tags.end_date)
```

Comment:
I wonder if we even need to normalize these dates. If a feature is tagged with a malformed date, `utilDatesOverlap()` would treat it as a blank date and assume it extends to the beginning or end of time. As a result, the feature would be shown no matter the selected date range. That would be an incentive to address the validator error.

Reply:
> wonder if we even need to normalize these dates

Good point. I have removed these, so `featureFitsDateRange()` passes them directly to `utilDatesOverlap()` and it does seem to work as expected.